plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt

```diff
@@ -367,8 +370,10 @@ class JKCodeBuilder(context: NewJ2kConverterContext) {
             }
         }
 
-
         override fun visitIfElseStatementRaw(ifElseStatement: JKIfElseStatement) {
+            if (ifElseStatement.hasLiftedReturn) {
+                printer.print("return ")
```

Comment:
This is very questionable. `return` statements should be represented as real JK elements, not printed out of thin air in the code builder.

Let's try to refactor this into a single conversion with an obvious algorithm. It should create a new return statement wrapping the if/while/try and detach all inner return statements (preserving the comments/formatting). If you meet some particular problem with this approach, let's discuss it with a concrete code example.